addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/handler/InnogyBridgeHandler.java

```diff
@@ -0,0 +1,843 @@
+            client.setOperationMode(capabilityId, autoMode);
+        } catch (Exception e) {
+            handleClientException(e);
+        }
+    }
+
+    /**
+     * Sends the command to set the dimm level of the {@link Device} with the given id. Is called by the
+     * {@link InnogyDeviceHandler} for {@link Device}s like ISD2 or PSD.
+     *
+     * @param deviceId
+     * @param dimLevel
+     */
+    public void commandSetDimmLevel(String deviceId, int dimLevel) {
+        try {
+            String capabilityId = deviceStructMan.getCapabilityId(deviceId, Capability.TYPE_DIMMERACTUATOR);
+            client.setDimmerActuatorState(capabilityId, dimLevel);
+        } catch (Exception e) {
+            handleClientException(e);
+        }
+    }
+
+    /**
+     * Sends the command to set the rollershutter level of the {@link Device} with the given id. Is called by the
+     * {@link InnogyDeviceHandler} for {@link Device}s like ISR2.
+     *
+     * @param deviceId
+     * @param rollerSchutterLevel
+     */
+    public void commandSetRollerShutterLevel(String deviceId, int rollerSchutterLevel) {
+        try {
+            String capabilityId = deviceStructMan.getCapabilityId(deviceId, Capability.TYPE_ROLLERSHUTTERACTUATOR);
+            client.setRollerShutterActuatorState(capabilityId, rollerSchutterLevel);
+        } catch (Exception e) {
+            handleClientException(e);
+        }
+    }
+
+    /**
+     * Handles all Exceptions of the client communication. For minor "errors" like an already existing session, it
+     * returns true to inform the binding to continue running. In other cases it may e.g. schedule a reinitialization of
+     * the binding.
+     *
+     * @param e the Exception
+     * @return boolean true, if binding should continue.
+     */
+    private boolean handleClientException(Exception e) {
+
+        // Session exists
+        if (e instanceof SessionExistsException) {
+            logger.info("Session already exists. Continuing...");
+            return true;
+
+            // Controller offline
+        } else if (e instanceof ControllerOfflineException) {
+            logger.error("innogy SmartHome Controller is offline.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, e.getMessage());
+            dispose();
+            scheduleReinitialize();
+            return false;
+
+            // Configuration error
+        } else if (e instanceof ConfigurationException) {
+            logger.error("Configuration error: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            dispose();
+            return false;
+
+            // invalid auth code
+        } else if (e instanceof InvalidAuthCodeException) {
+            logger.error("Error fetching access tokens. Invalid authcode! Please generate a new one. Detail: {}",
```

Comment:
As this is also only a initial error in case of a wrong binding configuration, I changed it to `warn` as well.